### PR TITLE
Exporter app map view + BBox creation functionality

### PR DESCRIPTION
**Pull request by alebinson**
Exporter app map view + BBox creation functionality

### CODE REVIEW

File: src/conflicts/components/map-container/map-wrapper.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  left: unset;
+}
+
+.mdc-dialog__title + .mdc-dialog__content{
+  padding-top: 20px;
+}
+
+.mdc-dialog__actions{
+  padding-right: 24px;
+}
```
Comment (asafMasa) [inline src/conflicts/components/map-container/map-wrapper.css:13]: end of line (add enter)

File: src/conflicts/components/map-container/polygon-selection-ui.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+const startDraw = jest.fn();
+const cancelDraw = jest.fn();
+const resetDraw = jest.fn();
+afterEach(() => {
+  startDraw.mockClear();
+  cancelDraw.mockClear();
+  resetDraw.mockClear();
+});
+
+// it('renders correctly', () => {
```
Comment (asafMasa) [inline src/conflicts/components/map-container/polygon-selection-ui.spec.tsx:16]: why comments?

File: src/conflicts/components/map-container/polygon-selection-ui.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+  const handleClose = (): void => {
+    setAnchorEl(null);
+  };
+
+  const [open, setOpen] = React.useState(false);
+
+  if (isSelectionEnabled) {
+    return (
+      <Tooltip content="Cancel the ongoing draw" align={'bottomLeft'}>
```
Comment (asafMasa) [inline src/conflicts/components/map-container/polygon-selection-ui.tsx:56]: shouldn't we use translate package for RTL langueges?

File: src/conflicts/components/map-container/polygon-selection-ui.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          isOpen={open}
+          onSetOpen={setOpen}
+          onPolygonUpdate={onPolygonUpdate}>
+        </DialogBBox>
+        <Menu
+          className={classes.fullWidth}
+          open={Boolean(anchorEl)}
+          onClose={handleClose}
+        >
+          {/* <MenuItem
```
Comment (asafMasa) [inline src/conflicts/components/map-container/polygon-selection-ui.tsx:87]: comment?

File: src/conflicts/components/map-container/polygon-selection-ui.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            onClick={(): void => {
+              onStartDraw(DrawType.BOX);
+              handleClose();
+            }}
+          >
+            Box
+          </MenuItem>
+          <MenuItem
+            onClick={(): void => {
+              // onStartDraw(DrawType.BOX);
```
Comment (asafMasa) [inline src/conflicts/components/map-container/polygon-selection-ui.tsx:105]: ?

File: package.json [hunk 6]
```diff
@@ -48,12 +48,14 @@
   "dependencies": {
     "@commitlint/cli": "^9.1.1",
     "@commitlint/config-conventional": "^9.1.1",
-    "@map-colonies/react-components": "1.0.1",
-    "@map-colonies/react-core": "1.0.1",
+    "@map-colonies/react-components": "file:../shared-components/packages/react-components",
```
[the remark this answers is not in the document]
Comment (alebinson) [reply package.json:51]: You are right, PR in review of Vector team https://github.com/MapColonies/shared-components/pull/34

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+  const classes = useStyle();
+  const formik = useFormik({
+    initialValues: {
+      bottomLeftLat: 0,
+      bottomLeftLon: 0,
+      topRightLat: 0,
+      topRightLon: 0,
+    },
+    onSubmit: values => {
+      // alert(JSON.stringify(values, null, 2));
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:78]: Sure it's still in working progress

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+      //       [34.70371442859832,32.022954156119006]
+      //   ]]
+      // };
+
+      const line = turf.lineString([
+        [values.bottomLeftLat, values.bottomLeftLon],
+        [values.topRightLat, values.topRightLon], 
+      ]);
+      const polygon = bboxPolygon(bbox(line));
+      console.log('polygon', polygon.geometry);
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:95]: All logs will be removed, still working on it

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+      //   coordinates:[[
+      //       [34.70371442859832,32.022954156119006],
+      //       [35.146463443653914,32.022954156119006],
+      //       [35.146463443653914,32.23608887564354],
+      //       [34.70371442859832,32.23608887564354],
+      //       [34.70371442859832,32.022954156119006]
+      //   ]]
+      // };
+
+      const line = turf.lineString([
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:90]: what do you mean?

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+    
+      onPolygonUpdate(polygon.geometry);
+    },
+  });
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <Dialog
+        open={isOpen}
+        onClose={evt => {
+          console.log(evt.detail.action);
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:105]: sure

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+          onSetOpen(false);
+        }}
+        onClosed={evt => console.log(evt.detail.action)}
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:108]: will be removed

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+          <DialogButton action="close" type="button">Cancel</DialogButton>
+          <DialogButton raised action="accept" type="submit" isDefaultAction>
+            Ok
+          </DialogButton>
+        </DialogActions>
+      </Dialog>
+    </form>
+
+  );
+}
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:165]: Done

File: src/conflicts/components/map-container/map-container.tsx [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+  const onReset = (): void => {
+    setSelectionPolygon(undefined);
+    props.handlePolygonReset();
+  };
+
+  return (
+    <div className={classes.map}>
+      <div className={`${classes.absolute}`}>
+        <div className={classes.filtersContainer}> {/*Paper*/}
```
[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/map-container.tsx:61]: will be removed

File: src/conflicts/components/map-container/map-wrapper.css (hunk 1 above)
Comment (alebinson) [reply src/conflicts/components/map-container/map-wrapper.css:13]: Done

File: src/conflicts/components/map-container/polygon-selection-ui.spec.tsx (hunk 2 above)
Comment (alebinson) [reply src/conflicts/components/map-container/polygon-selection-ui.spec.tsx:16]: still not implemented

File: src/conflicts/components/map-container/polygon-selection-ui.tsx (hunk 3 above)
Comment (alebinson) [reply src/conflicts/components/map-container/polygon-selection-ui.tsx:56]: Yes, will be in other PR of internacionalization

File: src/conflicts/components/map-container/polygon-selection-ui.tsx (hunk 4 above)
Comment (alebinson) [reply src/conflicts/components/map-container/polygon-selection-ui.tsx:87]: will be removed

File: src/conflicts/components/map-container/polygon-selection-ui.tsx (hunk 5 above)
Comment (alebinson) [reply src/conflicts/components/map-container/polygon-selection-ui.tsx:105]: will be removed

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 14]
```diff
@@ -1,18 +1,19 @@
-import React from 'react';
+import React, { useState } from 'react';
 import * as  turf from '@turf/helpers';
 import bbox from '@turf/bbox';
 import bboxPolygon from '@turf/bbox-polygon';
+import distance  from '@turf/distance/dist/js';
```
Comment (asafMasa) [inline src/conflicts/components/map-container/dialog-bbox.tsx:5]: @turf/distance why '/dist/js' extension?

File: src/conflicts/components/map-container/dialog-bbox.tsx [hunk 15]
```diff
@@ -49,13 +50,53 @@ import { Box } from '@map-colonies/react-components';
...
+
+interface BBoxCornersError {
+  latDistance: string;
+  lonDistance: string;
+}
+
+const validate = (values: BBoxCorners): BBoxCornersError => {
+  const errors: BBoxCornersError = {latDistance: '', lonDistance: ''};
+
+  const yDistance = distance(
```
Comment (asafMasa) [inline src/conflicts/components/map-container/dialog-bbox.tsx:81]: add try catch before all turf methods - they can throw exception

File: src/conflicts/components/map-container/dialog-bbox.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (asafMasa) [reply src/conflicts/components/map-container/dialog-bbox.tsx:119]: turf methods can throw exception for invalid data and our app to crush

[the remark this answers is not in the document]
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:119]: done

File: src/conflicts/components/map-container/dialog-bbox.tsx (hunk 14 above)
Comment (alebinson) [reply src/conflicts/components/map-container/dialog-bbox.tsx:5]: there is a problem with distance in current version, use feature alpha version

**Review by asafMasa — APPROVED** [review]

**Pull request merged**